conan/tools/cmake/cmakedeps/templates/macros.py

```diff
@@ -64,9 +64,37 @@ def template(self):
                    string(REGEX REPLACE "[^A-Za-z0-9.+_-]" "_" _LIBRARY_NAME ${_LIBRARY_NAME})
                    set(_LIB_NAME CONAN_LIB::${package_name}_${_LIBRARY_NAME}${config_suffix})
                    if(NOT TARGET ${_LIB_NAME})
+                       if(library_type STREQUAL "SHARED" AND is_host_windows)
+                         set(CMAKE_FIND_LIBRARY_SUFFIXES .dll ${CMAKE_FIND_LIBRARY_SUFFIXES})
+                         find_library(CONAN_SHARED_FOUND_LIBRARY NAMES ${_LIBRARY_NAME} PATHS ${package_bindir}
+                                      NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+                         if(NOT CONAN_SHARED_FOUND_LIBRARY)
+                           conan_message(STATUS "Cannot locate shared library: ${_LIBRARY_NAME}")
+                           set(CONAN_DLL_NOT_FOUND 1)
+                         endif()
+                       endif()
+
                        # Create a micro-target for each lib/a found
-                       add_library(${_LIB_NAME} UNKNOWN IMPORTED)
-                       set_target_properties(${_LIB_NAME} PROPERTIES IMPORTED_LOCATION ${CONAN_FOUND_LIBRARY})
+                       if (CONAN_SHARED_FOUND_LIBRARY)
+                           add_library(${_LIB_NAME} SHARED IMPORTED)
+                           set_target_properties(${_LIB_NAME} PROPERTIES IMPORTED_LOCATION ${CONAN_SHARED_FOUND_LIBRARY})
+                           set_target_properties(${_LIB_NAME} PROPERTIES IMPORTED_IMPLIB ${CONAN_FOUND_LIBRARY})
+                           conan_message(DEBUG "Found DLL and STATIC at ${CONAN_SHARED_FOUND_LIBRARY}, ${CONAN_FOUND_LIBRARY}")
+                        else()
+                           if(CONAN_DLL_NOT_FOUND)
```

Comment:
I'm trying to understand the if-else structure...

```cmake
    if (CONAN_SHARED_FOUND_LIBRARY)
        # ....
    else()
       if(CONAN_DLL_NOT_FOUND)
           # ....
       else()
           # ....
       endif()
       set_target_properties(${_LIB_NAME} PROPERTIES IMPORTED_LOCATION ${CONAN_FOUND_LIBRARY})
    endif()
```
Then if `library_type == "STATIC"` or not, it does not matter because `CONAN_DLL_NOT_FOUND` will always `0`, won't it?

Reply:
I think, if `library_type != "SHARED"` or "not windows" it will enter there.

Reply:
I agree that I also struggled a bit to follow the logic. Maybe it is worth to have 1 single level ``if-else`` structure starting from the ``if(library_type STREQUAL "SHARED" AND is_host_windows)``, even if there are more ``add_library()`` calls?

Reply:
I think now is much clear, thanks for the concerns and suggestions.